docs/versioning_vendor_api.md

```diff
@@ -0,0 +1,138 @@
+# Versioning Vendor API
+
+## Summary
+
+This document describes the steps required in order to increment the minor version of the Vendor API
+
+
+## Introducing new changes
+
+Changes in the API are specified in `VersionChange` classes and each version can be made up of one or more classes. Every change class must be configured in `app/lib/vendor_api.rb`
+
+A `VersionChange` class and has a `description`, an `action` and/or one or multiple resources set using the `resource` identifier.
+
+
+### Adding a new endpoint
+
+New endpoints are not by default made available via the API. In order to make them available from a specific version onwards, the endpoint's action must be configured in the version class and the version class must be mapped on a version.
```

Comment:
I'm curious how this interacts with `routes.rb`?

Reply:
We validate this using  a `constraints` on the vendor_api namespace. It checks that the controller action associated with a route is available in a version up to the one specified.

https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/routes.rb#L620 which checks through https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/initializers/route_validators.rb#L7 and more specifically https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/initializers/route_validators.rb#L28